### DISCUSSION TITLE
[4/1] LLMs, LLM Fine-Tuning

### DISCUSSION THREAD

**Discussion by tjdtnsu** [Pseudo-Lab/EfficientML#9] [General]
참고: Advanced Topics 제외
- LLM 소개
- SFT, RLHF, PEFT

**Comment by tjdtnsu** [comment]
- SFT: 다음 토큰 예측하는걸 파인 튜닝함
- RLHF: RL 기반 feedback 과정 추가하는 것. 여러 output을 샘플링해서 labeler가 ranking 매겨서 reward model 학습.
- DPO: 학습에 여러 모델이 관여하는데, 대신 하나의 SFT problem을 풀음. (reward 모델 학습 안함)
- BitFit: bias만 파인튜닝 하는것
- Adapter: Transformer layer 안에 FFN, nonlinearity로 이루어진 adapter layer 추가한것. latency는 증가함.
- Prompt tuning: 프롬프트 앞에 추가하는것. 첫 레이어에 넣은것. 모델크기 증가할수록 효과 좋음
- Prefix-Tuning: 각 레이어에 tunable prompt를 추가하는 것
- LoRA: trainable rank decompisiton을 각 layer마다 주입하는것. x@W+W@A@B인데, B는 0이므로 초기화는 0임  pretrained weight는 미학습. 이는 통합되어 latency 증가 없음
- QLoRA: base transformer를 4bit로 만들어버리는 것.
- Bit-Delta: 1bit 학습할 수 있게 만드는 것. 이것도 base model에 더하기로 됨
